tests/models/test_image.py

```diff
@@ -278,6 +279,65 @@ def test_image_coordinates():
     assert abs(image.gallat - 33.542) < 0.01
     assert abs(image.gallon - 160.922) < 0.01
 
+def test_image_cone_search( provenance_base ):
+    with SmartSession() as session:
+        image1 = None
+        image2 = None
+        image3 = None
+        image4 = None
+        try:
+            mjd = 60000.
+            endmjd = 60000.0007
+            kwargs = { 'mjd': 60000.,
+                       'end_mjd': 60000.0007,
+                       'format': 'fits',
+                       'exp_time': 60.48,
+                       'section_id': 'x',
+                       'project': 'x',
+                       'target': 'x',
+                       'instrument': 'x',
+                       'telescope': 'x',
+                       'filter': 'r',
+                      }
+            image1 = Image( 'one.fits', ra=120., dec=10., provenance=provenance_base, nofile=True, **kwargs )
+            image2 = Image( 'two.fits', ra=120.0002, dec=9.9998, provenance=provenance_base, nofile=True, **kwargs )
+            image3 = Image( 'three.fits', ra=120.0005, dec=10., provenance=provenance_base, nofile=True, **kwargs )
+            image4 = Image( 'four.fits', ra=60., dec=0., provenance=provenance_base, nofile=True, **kwargs )
+            session.add( image1 )
+            session.add( image2 )
+            session.add( image3 )
+            session.add( image4 )
+
+            sought = session.query( Image ).filter( Image.cone_search(120., 10., rad=1.02) ).all()
+            assert ( set( [ pathlib.Path( s.filepath ).name for s in sought ] )
+                     == { 'one.fits', 'two.fits' } )
```

Comment:
```suggestion
            assert image1.id in [im.id for im in sought]
            assert image.id in [im.id for im in sought]
```

Reply:
This has two advantages:
- You don't depend on other tests to clean up for this to work (i.e., imagine the database is not empty when you start this test). 
- You don't need to know the filenames for the images (which I intend to modify in the near future when we enforce having a file on disk before committing an image). 

Also, this is the way I've been testing these kinds of queries until now.

Reply:
Ok, that's true. But I still would test on the ids not on the filenames, and I would not assume the DB has no other images. This is a good way to make the tests break later on and never figure out why (when some other test leaves an Image in the DB and you suddenly - and only sometimes - have a failing test).